client/config.c

```diff
@@ -192,6 +242,32 @@ TDNFReadConfig(
         }
     }
 
+    dwError = TDNFAllocateMemory(UserAgentHeaderMaxLength, sizeof(char), (void**)&pszUserAgentHeader);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+	strcpy(pszUserAgentHeader, "User-Agent:tdnf/");
+    if (pConf->pszVarReleaseVer != NULL) {
+	    strcat(pszUserAgentHeader, pConf->pszVarReleaseVer);
+        strcat(pszUserAgentHeader, " ");
+    }
+    else
+        strcat(pszUserAgentHeader, "UNKNOWN ");
+
+    if (pConf->pszOSName != NULL)
+	    strcat(pszUserAgentHeader, pConf->pszOSName);
+    else
+        strcat(pszUserAgentHeader, "UNKNOWN_OS_NAME");
```

Comment:
Also here, and and just "unknown" is enough.

Reply:
use `snprintf` for this

Reply:
It should be `TDNFAllocateStringPrintf()`, see my comment above.